src/include/migraphx/check_shapes.hpp

```diff
@@ -165,6 +165,13 @@ struct check_shapes
         return *this;
     }
 
+    const check_shapes& not_tuple_type() const
```

Comment:
I would prefer to have a `allow_tuple_type` instead since the majority of operators dont allow tuples. This might require a `bool` flag that is checked in the destructor. Maybe we can put this in a seperate PR.

Reply:
Removed it for now.